### DISCUSSION TITLE
running on a method level

### DISCUSSION THREAD

**Discussion by SehamAAlharbi** [jplag/JPlag#1006] [Q&A] [answered]
Hi, 

I am using JPlag API and was wondering if I could let it run on a method level rather than a file level. how to do we adjust the JPlagOptions to make that?

thanks

**Comment by SehamAAlharbi** [comment]
Hi again, 

just another related question, is there a way for ignoring comments and white spaces? so they do not interfere with the similarity percentage? or this is a cleaning pre-process that should be done manually to get better results?

thanks in advance.

**Reply by tsaglam** [reply]
- Comments and whitespaces should not interfere with the similarity calculations, as this calculation is done on a token level. Whitespace should not affect plagiarism detection at all.
- To improve the results, you may adjust the minimum token match parameter, use basecode if you give your students a prescribed framework, and adjust the clustering if desired. There is no need for pre-processing, except maybe ensuring file format (should ideally be UTF-8).

Hope that helps!

**Reply by SehamAAlharbi** [reply]
Thanks, does this mean that JPlag will not identify commented-out code as a clone of active code?

**Reply by tsaglam** [reply]
No, comments are ignored; they are not extracted as tokens. See, for example, the token [types of the Java language module](https://github.com/jplag/JPlag/blob/main/languages/java/src/main/java/de/jplag/java/JavaTokenType.java). Note that the token types vary between language modules.

**Reply by SehamAAlharbi** [reply]
Thank you, that's really helpful. How about setting the threshold value? is there a recommended range of values?

**Reply by tsaglam** [reply]
If you mean `-m`, then I would say ignore it for now. The first parameter you should start to vary if your results are inconclusive is the min token match `-t`.  Increase it if you want less sensitivity, and decrease it if you want more sensitivity. Start from the default values, e.g., [9 for Java](https://github.com/jplag/JPlag/blob/284206ce766e5175cf4aeeddbded63494a105d0e/languages/java/src/main/java/de/jplag/java/Language.java#L41-L43).

**Comment by tsaglam** [comment, marked as answer]
Regarding your first question: 

> I am using JPlag API and was wondering if I could let it run on a method level rather than a file level.

What do you mean by method level? If you give your students a class with pre-defined empty methods, then you can use the basecode functionality (-bc). If you mean something different, please let me know.
